src/env-inl.h

```diff
@@ -185,14 +195,37 @@ inline Environment::Environment(v8::Local<v8::Context> context,
 }
 
 inline Environment::~Environment() {
+  CHECK_EQ(sub_worker_context_count(), 0);
   v8::HandleScope handle_scope(isolate());
 
+  PersistentHandleCleanup cleanup(this);
+  isolate()->VisitHandlesWithClassIds(&cleanup);
+
   context()->SetAlignedPointerInEmbedderData(kContextEmbedderDataIndex,
                                              nullptr);
 #define V(PropertyName, TypeName) PropertyName ## _.Reset();
   ENVIRONMENT_STRONG_PERSISTENT_PROPERTIES(V)
 #undef V
   isolate_data()->Put();
+
+  if (using_cares_)
+    ares_destroy(ares_channel());
+
+  // For valgrind. The main environment cannot deal with CleanupHandles()
+  // for some reason.
```

Comment:
Sure but it's not related to this PR - the CleanupHandles method of environment was never called anywhere normally (despite some comment saying so) and IIRC when I tried it, something broke. So I just delete the queued handles (for workers this queue is always empty at this point) for now to avoid valgrind reports for them.